libraries/ms-helper/src/mcp2515.c

```diff
@@ -31,19 +32,29 @@ static const Mcp2515RxBuffer s_rx_buffers[] = {
   { .id = MCP2515_READ_RXB1SIDH, .data = MCP2515_READ_RXB1D0, .int_flag = MCP2515_CANINT_RX1IE },
 };
 
+static const uint8_t s_brp_lookup[NUM_MCP2515_BITRATES] = {
+  // BRP calculated for different bitrates
+  [MCP2515_BITRATE_125KBPS] = MCP2515_CAN_BRP_125KBPS,
+  [MCP2515_BITRATE_250KBPS] = MCP2515_CAN_BRP_250KBPS,
+  [MCP2515_BITRATE_500KBPS] = MCP2515_CAN_BRP_500KBPS,
+};
+
 // SPI commands - See Table 12-1
 static void prv_reset(Mcp2515Storage *storage) {
   uint8_t payload[] = { MCP2515_CMD_RESET };
   spi_exchange(storage->spi_port, payload, sizeof(payload), NULL, 0);
+  delay_us(100);
 }
 
 static void prv_read(Mcp2515Storage *storage, uint8_t addr, uint8_t *read_data, size_t read_len) {
+  CRITICAL_SECTION_AUTOEND;
```

Comment:
what's this for? do we need it?

Reply:
Sets the function as critical, disabling interrupts. Removed for now, we can add it back if there are problems down the line.